### PR TITLE
Toggle indoor/outdoor button

### PR DESCRIPTION
**Pull request by gitjul**
1. add info why not showing indoor sessions - @paulinamaterna needs styling!
2. change structure of toggle keeping accessibility in mind. use buttons instead of checkbox.
3. ooops! accidentally already styled the toggle buttons ;) @paulinamaterna check it out you may want to fix/reorganize/improve

55415346-25791800-556d-11e9-804e-6d18a2ad540e

### CODE REVIEW

File: app/javascript/elm/src/Main.elm [hunk 1]
```diff
@@ -6,6 +6,7 @@ import Browser.Navigation
 import Data.Session exposing (..)
 import Html exposing (Html, a, button, dd, div, dl, dt, form, h2, h3, img, input, label, li, main_, nav, p, span, text, ul)
 import Html.Attributes as Attr
+import Html.Attributes.Aria exposing (..)
```
[the remark this answers is not in the document]
Comment (gitjul) [reply app/javascript/elm/src/Main.elm]: that's true.

File: app/javascript/elm/src/Main.elm [hunk 2]
```diff
@@ -331,7 +332,8 @@ view model =
                         text ""
                     , div [ Attr.class "map-container" ]
                         [ if model.isIndoor && not model.isHttping then
-                            div [ Attr.class "overlay" ] []
+                            div [ Attr.class "overlay" ]
+                                [ div [ Attr.class "change-this-classname-Pina" ] [ text "In order to preserve privacy, we don't display indoor sessions on the map." ] ]
```
[the remark this answers is not in the document]
Comment (gitjul) [reply app/javascript/elm/src/Main.elm]: good point, will ask Michael & Pina

File: app/javascript/elm/src/Main.elm [hunk 3]
```diff
@@ -531,17 +533,36 @@ viewFixedFilters model =
...
+            [ Attr.type_ "button"
+            , if not requirement then
+                Attr.class "toggle-button toggle-button--pressed"
+
+              else
+                Attr.class "toggle-button"
+            , ariaLabel label2
+            , Events.onClick (callback False)
             ]
+            [ text label2 ]
```
[the remark this answers is not in the document]
Comment (gitjul) [reply app/javascript/elm/src/Main.elm]: exactly, that last part was what I was missing when I was thinking of doing that. thanks!!

File: app/javascript/elm/src/Main.elm [hunk 4]
```diff
@@ -531,17 +533,36 @@ viewFixedFilters model =
...
+        , label [] [ text "type" ]
+        , viewToggleButtons "indoor" "outdoor" model.isIndoor ToggleIndoor
+        ]
+
+
+viewToggleButtons : String -> String -> Bool -> (Bool -> Msg) -> Html Msg
+viewToggleButtons label1 label2 requirement callback =
+    div []
+        [ button
+            [ Attr.type_ "button"
```
[the remark this answers is not in the document]
Comment (gitjul) [reply app/javascript/elm/src/Main.elm]: you don't have to in this case, but it won't hurt. basically there are 3 types of buttons, and it's most important to add that `type=button` when it's inside a form and it's not submitting the form. if you don't the default type of the button will be `submit`. https://html.com/attributes/button-type/

File: app/javascript/elm/src/Main.elm (hunk 1 above)
[the remark this answers is not in the document]
Comment (3v0k4) [reply app/javascript/elm/src/Main.elm]: The awesome thing about having a compiler is that you could theoretically create an html library that compiles only if you pass all of the accessibility thingies when creating dom nodes. The awesomer thing is that the library exists https://package.elm-lang.org/packages/tesk9/accessible-html/latest/

**Comment by 3v0k4** [comment]
Good job @gitjul !

**Pull request merged**